packages/arb-token-bridge-ui/src/state/cctpState.ts

```diff
@@ -498,13 +482,73 @@ export function useCctpFetching({
     }
   }, [withdrawals, setTransfers])
 
+  const setPendingTransfer = useCallback(
+    (transfer: PartialMergedTransaction, isDeposit: boolean) => {
+      const mutate = isDeposit ? mutateDeposits : mutateWithdrawals
+      mutate(
+        {
+          pending: [transfer as MergedTransaction],
+          completed: []
+        },
+        {
+          populateCache(result, currentData) {
+            if (!currentData) {
```

Comment:
Can we use a simpler syntax and a simplified logic here. ie.   simple `mutate(key, data, options)` and appending the newly received tx to data and not using `options.populateCache` here?

Reply:
I need to try again. But I think the issue with `mutate` was the merging (we don't want to override)

Reply:
I had to revert the changes. Using `populateCache` guarantee us that `(result, currentData)` are up to date. We don't have this guarantee with manually using `mutate`